src/ReservoirPair.sol

```diff
@@ -357,32 +361,32 @@ abstract contract ReservoirPair is ReservoirERC20 {
         assetManager.afterLiquidityEvent();
     }
 
-    function adjustManagement(int256 token0Change, int256 token1Change) external {
+    function adjustManagement(int256 aToken0Change, int256 aToken1Change) external {
         require(msg.sender == address(assetManager), "AMP: AUTH_NOT_MANAGER");
-        require(token0Change != type(int256).min && token1Change != type(int256).min, "AMP: CAST_WOULD_OVERFLOW");
+        require(aToken0Change != type(int256).min && aToken1Change != type(int256).min, "AMP: CAST_WOULD_OVERFLOW");
```

Comment:
technically we don't need this check, since the cast would overflow if `type(int256).min` is supplied. 

I know cuz I tried it

Reply:
If you cast int256.min to uint256 do you get a big number? or how/where does this get caught?

Reply:
it gets caught at 

```
uint104 lDelta = uint104(uint256(-aToken0Change));
```

Specifically the negating operation. When we negate `aToken0Change`, it attempts to assign a value in the same type, but with opposing signs. And since `type(int256).max` (....9967) is 1 less than `type(int256).min` (-...9968), this casting will fail. 

For the positive case, the cast will always succeed without problems.

Reply:
Are you sure the cast will fail, that isnt what i would expect. I thought solidity casts are unchecked?

Reply:
Technically it isn't casting that fails, it's the negation and assigning of a value that doesn't fit. If you only cast it without negating, it will not fail. I'm sure of this because I've written tests in my own repo to test it out. 

You can run it yourself [here](https://github.com/xenide/balancer-log-math/blob/881860c9260c320e9f1fbf64a48808b50516edf6/test/Counter.t.sol#L43). This behavior only started in 0.8.0 onwards.

Reply:
Interesting, ill update my understanding haha. So safeCast is only needed for uint to uint or int to int casts?

Reply:
correct `SafeCast` is most useful when you're casting from a wider type to a narrower type.  

When casting from an `int` to a `uint`, and vice versa, we better check the sign. Although there're functions such as `toUint256(int256 value)` and `toInt256(uint256 value)` in the lib, the onus is still on us to check the sign since it reverts if the types don't fit. 

In our case, since we check if `token0Change >/< 0 `, our unchecked casting to `uint256` is **safe**. 

Therefore, I've deleted the `require` statement.